### ISSUE TITLE
feat(model): emit bounded V4.1 attention and MoE activation checkpoints

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13325]
# feat(model): emit bounded V4.1 attention and MoE activation checkpoints

```routing
repo: fak
lane: model
paths: ["internal/model/v41_forward.go","internal/model/v41_activation_trace.go","internal/model/v41_activation_trace_test.go"]
expected_steps: 3
```

## Working spine
Expose V4.1 integration failures before another full-model hardware run, using existing runtime and test surfaces.

## Current state
V4.1 has named stages and layer errors (v41_forward.go:60-123), and computetrace already has an opt-in global recorder. There is no bounded layer-stage activation producer binding the real V4.1 forward to a first-divergence numerical comparison.

## Why this is next
The native mission has reached a token but continues through expensive sequential bring-up. This leaf removes one class of uncertainty in a bounded software check while the current performance work proceeds.

## Parent context
anthony-chaudhary/fak#12640; engineering support for anthony-chaudhary/fak#13294 and MISSION-halo-v41-native. Existing deepseek-v41-flash and public issue missions can dispatch this single-package leaf.

## Core through-line
Use the preceding computetrace sample contract at exactly four existing boundaries per layer: qLat after the optional q RMSNorm and before wq_b (v41_forward.go:1946); kv after optional kv RMSNorm and before RoPE (:1989); attnOut[t] after grouped output projection (:2154); and moe after v41SharedExpertAdd but before delta/residual mixing (:2201). Use distinct q_latent, kv_latent, attention_projected and moe_sum keys, never a concatenation with ambiguous layout. Add a small helper that emits deterministic logical sample indices only when tracing is enabled; preserve backend identity and bind config, input-ID and weight-fixture provenance without recording raw prompt text or full weights. Exercise the actual Model.Forward path and compare its trace to independently authored tiny fixtures with CompareActivationTraces. Record pinned source/config/seed/dtype/input/weight digests in fixture metadata. A separate test-author context must produce expected samples without production helpers. Reuse available model identity; if a trustworthy weight identity is unavailable, mark evidence incomplete instead of hashing a full checkpoint or inventing an identity.

## Gold-plating boundary
Four boundary identities only; a single in-flight Forward owns the existing global recorder. No per-kernel dump, tensor persistence service, new user flag, checkpoint loader changes, GPU synchronization or change to forward numerical behavior. Do not copy complete rows before slicing; samples must respect the recorder caps.

## Worker packet
DS4.1 or Qwen3.8 capability level; S1. Own only the paths below. Three steps: establish the specified failing/missing contract with an independent fixture; implement only the bounded change; run the named and owning-package gates and land with explicit paths. No open-ended study or architecture choice remains. If a source symbol moved, locate its equivalent; do not silently widen the contract.

## Work unit
leaf

## Assumptions
- The pinned V4.1 architecture and existing production/test seams remain the authority; re-read changed symbols before editing.

## Confusion risks
- A passing synthetic witness proves the declared software envelope only, not physical performance or complete checkpoint equivalence.

## Coordination
- Acquire the listed public lane and preserve peer work. Serialize shared model-lane edits; use the dependencies section for start holds.

## Trigger
Operator-requested architecture planning after repeated sequential native bring-up failures; source-backed gaps documented in Current state.

## Batch policy
One issue per unique fak-native-shiftleft-key; search open and closed tickets before filing or expanding. Update the existing leaf on rerun.
## Work scale
S1

## Model tier
Required: T1
Optimal: T1

## Dependencies and scope fences
- Start blocked by: anthony-chaudhary/fak#13324 (bounded activation comparison contract).
- Coordinates with: anthony-chaudhary/fak#13299 (timing is separate from numeric samples).

## Reference and reuse decision
ADAPT SGLang's comparison-at-stage-boundaries method; reuse existing computetrace.Enable/Enabled/Record and V4.1 typed stage vocabulary. The independent latent-norm fixture is useful evidence but not an implementation dependency.

## Reproduction witness
Source-backed coverage gap: inspect the functions and current tests named above at public revision 3ce59641b971a31b6bf19838290e64c0bf5d1033, then add the named regression. A missing test is not a passing baseline. Capture its RED result or explicit missing-coverage finding before changes; do not claim an unexecuted production failure.

## Done condition / witness
Witness: `go test ./internal/model -run '^TestV41ActivationTrace$' -count=1 -v`
The test is a required deliverable, not asserted to exist today. Confirm the exact test appears in `go test ./internal/model -list '^TestV41ActivationTrace$'`; empty matches and unexpected skips fail acceptance. Then run `go test ./internal/model -count=1` and required boundary gates after the last edit, in the provisioned public candidate worktree.

## Verifiable Witness
```
go test ./internal/model -run '^TestV41ActivationTrace$' -count=1 -v
```

## Definition of done
- [ ] A real tiny Forward emits ordered layer/stage records with expected shape/provenance; an independently perturbed expected attention checkpoint is reported as the first sampled divergence.
- [ ] Tracing disabled produces identical logits and zero added trace allocations; enabled tracing respects the fixed caps and reports truncation as incomplete evidence.
- [ ] Focused and model/computetrace package gates pass; the existing tracing activation mechanism reaches these production callbacks and no physical qualification is claimed.

## Acceptance gate
Every checklist item is required; focused tests plus owning-package tests and independent review must pass. This is [SW-VERIFIED] coverage only. Physical throughput and coherent two-prompt qualification remain open with the existing mission owner.

## Blast radius and fallback
One package: internal/model. Existing behavior remains the fallback until this leaf passes. Preserve unrelated work, fail closed on unsupported shapes, and retain the prior physical mission gate. No new hardware run is a prerequisite.

## Closure binding
DCO sign-off, explicit paths, issue reference, and (fak model) commit subject suffix. Follow existing provenance and lease rules. Land via Fak Flow and push; no agent-created PR. Closure covers this leaf only.

## Likely files
- internal/model/v41_forward.go
- internal/model/v41_activation_trace.go
- internal/model/v41_activation_trace_test.go

## Expected steps
3

## Value
- Centrality: Core
- P1 Context: advanced - turns a known integration gap into a reusable local check
- P2 Net value: advanced - avoids repeated full-model diagnosis without relaxing correctness
- P3 Adaptation: preserved - reuses existing contracts and keeps runtime policy stable
- P4 Operations: advanced - bounded inputs and exact failure evidence support ordinary workers

## Work estimate
Estimate: 3 points

## Overall completion contribution
Contribution: 3/30 points

## Completion standard
development

## Qualification target
Pinned V4.1 text architecture; scoped synthetic or metadata-only software gate on a CPU development host.

## Planning baseline
Source inspection and planning only at filing; implementation and its tests are not yet witnessed.

**Comment by anthony-chaudhary** [comment]
Landed at 16240942a9bf (pushed to main).

Witness (re-run on landed trunk):
`
go test ./internal/model -run '^TestV41ActivationTrace\$' -count=1 -v
--- PASS: TestV41ActivationTrace (0.08s)
    --- PASS: TestV41ActivationTrace/disabled_records_nothing_and_preserves_logits
    --- PASS: TestV41ActivationTrace/enabled_emits_ordered_stage_records_with_provenance_and_shape
    --- PASS: TestV41ActivationTrace/a_repeat_forward_is_EQUAL_to_the_captured_reference
    --- PASS: TestV41ActivationTrace/an_independently_perturbed_attention_checkpoint_is_the_first_divergence
    --- PASS: TestV41ActivationTrace/cap_truncation_is_incomplete_evidence,_never_agreement
PASS
`

Diff: \internal/model/v41_activation_trace.go\ (new producer + provenance), \internal/model/v41_activation_trace_test.go\ (new witness), \internal/model/v41_forward.go\ (four boundary emits threaded through v41Layer).

Evidence class: [SW-VERIFIED] bounded software coverage only. A passing synthetic witness proves the declared software envelope, not physical performance or complete checkpoint equivalence. Physical throughput / two-prompt coherent-output qualification remain open with the existing mission owner (#12640, private #1395).

Fixes #13325
